pkg/services/ngalert/notifier/file_store.go

```diff
@@ -85,10 +85,20 @@ func (fs *FileStore) Persist(ctx context.Context, filename string, st State) (in
 	return int64(len(bytes)), err
 }
 
+// Delete will remove the file from the filestore.
+func (fs *FileStore) Delete(ctx context.Context, filename string) error {
+	if fs.IsExists(filename) {
+		if err := os.Remove(fs.pathFor(filename)); err != nil {
+			return err
+		}
+	}
+	return fs.kv.Del(ctx, filename)
+}
+
 // IsExists verifies if the file exists or not.
 func (fs *FileStore) IsExists(fn string) bool {
 	_, err := os.Stat(fs.pathFor(fn))
-	return os.IsExist(err)
+	return !os.IsNotExist(err)
```

Comment:
`os.Stat` does not return an `fs.ErrExist` if the file exists, it will just return `nil` in the error field. So we need to use `IsNotExist` to check if the error is of type `fs.ErrNotExist` which is returned by `os.Stat` if the given path/file is not existing.

Reply:
@yuri-tceretian the new test I introduced catches the bug, this is how I found it in the first place 😄  Good catch regarding `errors`, I'll change that.